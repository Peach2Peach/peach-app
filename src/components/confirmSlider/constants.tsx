import tw from '../../styles/tailwind'

export const padding = tw`w-1`.width as number
export const trackWidth = 260
export const knobWidth = tw`w-18`.width as number
