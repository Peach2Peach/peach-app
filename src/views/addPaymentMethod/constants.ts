import { CurrencyType } from "../../store/offerPreferenes/types";

export const defaultCurrencies: Record<CurrencyType, Currency> = {
  europe: "EUR",
  latinAmerica: "ARS",
  africa: "ZAR",
  global: "USDT",
  asia: "AED",
  northAmerica: "CAD",
  middleEast: "ILS",
  oceania: "AUD",
};
