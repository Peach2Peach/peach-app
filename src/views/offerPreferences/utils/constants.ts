const horizontalPaddingForSlider = 8
export const trackMin = horizontalPaddingForSlider
