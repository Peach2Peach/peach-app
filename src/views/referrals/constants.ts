export const POINTS_PER_FREE_TRADE = 100;
export const NUMBER_OF_FREE_TRADES = 2;
export const REWARDINFO: Reward[] = [
  { id: "customReferralCode", requiredPoints: 100 },
  {
    id: "noPeachFees",
    requiredPoints: POINTS_PER_FREE_TRADE * NUMBER_OF_FREE_TRADES,
  },
  { id: "sats", requiredPoints: 300 },
];
