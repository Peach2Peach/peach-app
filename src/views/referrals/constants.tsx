export const REWARDINFO: Reward[] = [
  { id: 'customReferralCode', requiredPoints: 100 },
  { id: 'noPeachFees', requiredPoints: 200 },
  { id: 'sats', requiredPoints: 300 },
]
