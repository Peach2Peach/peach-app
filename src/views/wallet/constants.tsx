export const WALLETS: Record<Chain, keyof HomeTabParamList> = {
  bitcoin: "wallet",
  lightning: "lightningWallet",
  liquid: "liquidWallet",
};
