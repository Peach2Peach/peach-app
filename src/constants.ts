import { getBuildNumber, getUniqueId, getVersion, isEmulatorSync } from 'react-native-device-info'
import { unique } from './utils/array'
import { sha256 } from './utils/crypto/sha256'

export const SATSINBTC = 100000000
export const MSINADAY = 86400000

export let PEACHPGPPUBLICKEY = ''
export const setPeachPGPPublicKey = (pgpPublicKey: string) => (PEACHPGPPUBLICKEY = pgpPublicKey)

export let PEACHFEE = 0.015
export const setPeachFee = (fee: number) => (PEACHFEE = fee)

export const MAXMININGFEE = 20000

// time go automatically restart app when calling app from background after this time has passed
export const TIMETORESTART = 1000 * 60 * 5

export const APPVERSION = getVersion()
export const BUILDNUMBER = getBuildNumber()

export let CLIENTSERVERTIMEDIFFERENCE = 0
export const setClientServerTimeDifference = (diff: number) => (CLIENTSERVERTIMEDIFFERENCE = diff)

export const ISEMULATOR = isEmulatorSync()

export const UNIQUEID = sha256(getUniqueId())

export let MINAPPVERSION = APPVERSION
export const setMinAppVersion = (ver: string) => (MINAPPVERSION = ver)

export let LATESTAPPVERSION = APPVERSION
export const setLatestAppVersion = (ver: string) => (LATESTAPPVERSION = ver)

export let CURRENCIES: Currency[] = ['EUR', 'CHF', 'GBP', 'SEK']

export let COUNTRIES: Country[] = ['DE', 'FR', 'IT', 'ES', 'NL', 'UK', 'SE']

export let PAYMENTMETHODS: PaymentMethod[] = ['sepa']
export let PAYMENTMETHODINFOS: PaymentMethodInfo[] = [
  {
    id: 'sepa',
    currencies: ['EUR'],
    annonymous: false,
  },
]

export const PAYMENTCATEGORIES: PaymentCategories = {
  bankTransfer: ['sepa'],
  onlineWallet: ['paypal', 'revolut', 'wise', 'twint', 'swish'],
  giftCard: ['giftCard.amazon'].concat(COUNTRIES.map((c) => `giftCard.amazon.${c}`)) as PaymentMethod[],
  localOption: ['mbWay', 'bizum', 'satispay'],
  cash: ['cash'],
  cryptoCurrency: [],
}

export const LOCALPAYMENTMETHODS: LocalPaymentMethods = {
  EUR: {
    IT: ['satispay'],
    PT: ['mbWay'],
    ES: ['bizum'],
  },
}

export const APPLINKS: Partial<Record<PaymentMethod, { appLink?: string; url: string; userLink?: string }>> = {
  paypal: {
    url: 'https://paypal.com/open_web',
    userLink: 'https://paypal.com/paypalme/',
  },
  revolut: {
    url: 'https://revolut.com/app',
    userLink: 'https://revolut.me/',
  },
  satispay: { url: 'https://satispay.com/app' },
  wise: { url: 'https://wise.com/user/account' },
  'giftCard.amazon.DE': { url: 'https://www.amazon.de/dp/B0B2Q4ZRDW' },
  'giftCard.amazon.FR': { url: 'https://www.amazon.fr/dp/B004MYH1YI' },
  'giftCard.amazon.IT': { url: 'https://www.amazon.it/dp/B07DWRCL44' },
  'giftCard.amazon.ES': { url: 'https://www.amazon.es/dp/B07H8STZ9N' },
  'giftCard.amazon.NL': { url: 'https://www.amazon.nl/dp/B07W6D728D' },
  'giftCard.amazon.UK': { url: 'https://www.amazon.co.uk/dp/B07S6C1DZ6' },
  'giftCard.amazon.SE': { url: 'https://www.amazon.se/dp/B089VNKFM7' },
}

export const setPaymentMethods = (paymentMethodInfos: PaymentMethodInfo[]) => {
  PAYMENTMETHODINFOS = paymentMethodInfos
  CURRENCIES = paymentMethodInfos.reduce((arr, info) => arr.concat(info.currencies), [] as Currency[]).filter(unique())
  COUNTRIES = paymentMethodInfos
    .reduce((arr, info) => arr.concat(info.countries || []), [] as Country[])
    .filter(unique())
  PAYMENTMETHODS = paymentMethodInfos.map((method) => method.id)
}

export let MINTRADINGAMOUNT = 200000
export let MAXTRADINGAMOUNT = 5000000
export const setMinTradingAmount = (amount: number) => (MINTRADINGAMOUNT = amount)
export const setMaxTradingAmount = (amount: number) => (MAXTRADINGAMOUNT = amount)

type Timers = {
  [key in ContractAction]: number
}

// Time in ms
export const TIMERS: Timers = {
  none: 0,
  kycResponse: 1000 * 60 * 60 * 12,
  sendPayment: 1000 * 60 * 60 * 12,
  confirmPayment: 1000 * 60 * 60 * 12,
}

// Reputation
export const GOLDMEDAL = 0.9
export const SILVERMEDAL = 0.7

export const MEDALS = ['fastTrader', 'superTrader', 'ambassador']
