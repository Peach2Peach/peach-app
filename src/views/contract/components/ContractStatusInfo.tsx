import { ReactElement } from 'react'
import { View } from 'react-native'
import { Icon, Text, Timer } from '../../../components'
import tw from '../../../styles/tailwind'
import { getPaymentExpectedBy } from '../../../utils/contract'
import i18n from '../../../utils/i18n'
import { shouldShowConfirmCancelTradeRequest } from '../../../utils/overlay'

type ContractStatusInfoProps = {
  contract: Contract
  requiredAction: ContractAction
  view: ContractViewer
}
export const ContractStatusInfo = ({ contract, requiredAction, view }: ContractStatusInfoProps): ReactElement => {
  if (contract.disputeActive) return (
    <View style={tw`flex-row items-center justify-center`}>
      <Text style={tw`text-center button-medium`}>
        {i18n('contract.disputeActive') + ' - ' + i18n('contract.checkTheChat')}
      </Text>
    </View>
  )
  if (shouldShowConfirmCancelTradeRequest(contract, view)) return (
    <View style={tw`flex-row items-center justify-center`}>
      <Text style={tw`text-center button-medium text-warning-dark-2`}>
        {i18n('offer.requiredAction.confirmCancelation')}
      </Text>
    </View>
  )

  if (contract.cancelationRequested) return (
    <View style={tw`flex-row items-center justify-center`}>
      <Text style={tw`text-center button-medium`}>{i18n('contract.cancelationPending')}</Text>
      <Icon id="xCircle" style={tw`w-5 h-5 ml-1 -mt-0.5`} color={tw`text-black-3`.color} />
    </View>
  )
  if (requiredAction === 'sendPayment') {
    const paymentExpectedBy = getPaymentExpectedBy(contract)
    if (Date.now() < paymentExpectedBy) {
      return <Timer text={i18n(`contract.timer.${requiredAction}.${view}`)} end={paymentExpectedBy} />
    }
    return (
      <View style={tw`flex-row items-center`}>
        <Text style={tw`uppercase button-medium`}>{i18n('contract.timer.paymentTimeExpired.buyer')}</Text>
        <Icon id={'clock'} style={tw`w-5 h-5 ml-1`} />
      </View>
    )
  }
  if (view === 'buyer' && requiredAction === 'confirmPayment') return (
    <View style={tw`flex-row items-center justify-center`}>
      <Text style={tw`text-center button-medium`}>{i18n('contract.timer.confirmPayment.buyer')}</Text>
    </View>
  )
  if (view === 'seller' && requiredAction === 'confirmPayment') return (
    <View style={tw`flex-row items-center justify-center`}>
      <Text style={tw`text-center button-medium`}>{i18n('contract.timer.confirmPayment.seller')}</Text>
      <Icon id="check" style={tw`w-5 h-5 ml-1 -mt-0.5`} color={tw`text-success-main`.color} />
    </View>
  )
  return <></>
}
