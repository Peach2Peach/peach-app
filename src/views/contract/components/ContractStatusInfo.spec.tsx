import { createRenderer } from 'react-test-renderer/shallow'
import { contract } from '../../../../tests/unit/data/contractData'
import { MSINADAY } from '../../../constants'
import { ContractStatusInfo } from './ContractStatusInfo'

const now = new Date('2020-01-01')
jest.useFakeTimers({ now })

const useContractContextMock = jest.fn()
jest.mock('../context', () => ({
  useContractContext: () => useContractContextMock(),
}))

describe('ContractStatusInfo', () => {
  const renderer = createRenderer()

  it('renders correctly for buyer when requiredAction is sendPayment and paymentExpectedBy is in the future', () => {
    const activeContract = { ...contract, paymentExpectedBy: new Date(now.getTime() + MSINADAY) }
    useContractContextMock.mockReturnValueOnce({ contract: activeContract, view: 'buyer' })

    renderer.render(<ContractStatusInfo requiredAction="sendPayment" />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })

  it('renders correctly for buyer when requiredAction is sendPayment and paymentExpectedBy is in the past', () => {
    const pastContract = { ...contract, paymentExpectedBy: new Date(now.getTime() - MSINADAY) }
    useContractContextMock.mockReturnValueOnce({ contract: pastContract, view: 'buyer' })

    renderer.render(<ContractStatusInfo requiredAction="sendPayment" />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })

  it('renders correctly for buyer when requiredAction is confirmPayment', () => {
    useContractContextMock.mockReturnValueOnce({ contract, view: 'buyer' })

    renderer.render(<ContractStatusInfo requiredAction="confirmPayment" />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })

  it('renders correctly for seller when requiredAction is confirmPayment', () => {
    useContractContextMock.mockReturnValueOnce({
      contract,
      view: 'seller',
    })

    renderer.render(<ContractStatusInfo requiredAction="confirmPayment" />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })

  it('renders correctly when contract is in dispute', () => {
    const disputeContract = { ...contract, disputeActive: true }
    useContractContextMock.mockReturnValueOnce({ contract: disputeContract })

    renderer.render(<ContractStatusInfo requiredAction="sendPayment" />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })

  it('renders correctly for seller when cancellation is pending', () => {
    const cancellationContract = { ...contract, cancelationRequested: true }
    useContractContextMock.mockReturnValueOnce({ contract: cancellationContract })

    renderer.render(<ContractStatusInfo requiredAction="sendPayment" />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })

  it('renders correctly when cancel trade request confirmation is required from buyer', () => {
    const cancelTradeRequestContract = { ...contract, cancelationRequested: true }
    useContractContextMock.mockReturnValueOnce({ contract: cancelTradeRequestContract })

    renderer.render(<ContractStatusInfo requiredAction="sendPayment" />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })

  it('renders correctly when payment method is cash trade', () => {
    const cashTradeContract: Contract = { ...contract, paymentMethod: 'cash.de.berlin' }
    useContractContextMock.mockReturnValueOnce({ contract: cashTradeContract })

    renderer.render(<ContractStatusInfo requiredAction="sendPayment" />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
})
