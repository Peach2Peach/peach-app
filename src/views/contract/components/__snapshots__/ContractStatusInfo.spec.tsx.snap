// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContractStatusInfo renders correctly for buyer when requiredAction is confirmPayment 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <PeachText
    style={
      {
        "fontFamily": "Baloo2-SemiBold",
        "fontSize": 14,
        "letterSpacing": 0.5,
        "lineHeight": 24,
        "textAlign": "center",
        "textTransform": "uppercase",
      }
    }
  >
    waiting for confirmation...
  </PeachText>
</View>
`;

exports[`ContractStatusInfo renders correctly for buyer when requiredAction is sendPayment and paymentExpectedBy is in the future 1`] = `
<Timer
  end={1577923200000}
  text="you should pay within"
/>
`;

exports[`ContractStatusInfo renders correctly for buyer when requiredAction is sendPayment and paymentExpectedBy is in the past 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <PeachText
    style={
      {
        "fontFamily": "Baloo2-SemiBold",
        "fontSize": 14,
        "letterSpacing": 0.5,
        "lineHeight": 24,
        "textTransform": "uppercase",
      }
    }
  >
    payment too late!
  </PeachText>
  <Icon
    id="clock"
    style={
      {
        "height": 20,
        "marginLeft": 4,
        "width": 20,
      }
    }
  />
</View>
`;

exports[`ContractStatusInfo renders correctly for seller when cancellation is pending 1`] = `<React.Fragment />`;

exports[`ContractStatusInfo renders correctly for seller when requiredAction is confirmPayment 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <PeachText
    style={
      {
        "fontFamily": "Baloo2-SemiBold",
        "fontSize": 14,
        "letterSpacing": 0.5,
        "lineHeight": 24,
        "textAlign": "center",
        "textTransform": "uppercase",
      }
    }
  >
    buyer made payment
  </PeachText>
  <Icon
    color="#05A85A"
    id="check"
    style={
      {
        "height": 20,
        "marginLeft": 4,
        "marginTop": -2,
        "width": 20,
      }
    }
  />
</View>
`;

exports[`ContractStatusInfo renders correctly when cancel trade request confirmation is required from buyer 1`] = `<React.Fragment />`;

exports[`ContractStatusInfo renders correctly when contract is in dispute 1`] = `<React.Fragment />`;

exports[`ContractStatusInfo renders correctly when payment method is cash trade 1`] = `<React.Fragment />`;
